teps/0003-tekton-catalog-organization.md

```diff
@@ -0,0 +1,465 @@
+
+* The root OWNERS file maps the tektoncd/catalog-maintainers team. This team is responsible for the overall catalog, from maintaining Resources (especially those who do not have a specific set of OWNERS), making sure the testing infrastructure is healthy, triaging issues, …
+
+* The *per-resource *OWNERS are responsible for maintaining their Resource (updating images, behaviour, …)
+
+### Organization
+
+Based on the previous (Ownership, Versionning, Compatibility), the following organization is proposed.
+
+```bash
+# Optional: owner(s) and reviewer(s) of the resource (all versions)
+./{resource-type}/{resource-name}/OWNERS
+# The README of the resource (what is the resource about, list of versions, …)
+./{resource-type}/{resource-name}/README.md
+
+# The resource itself
+./{resource-type}/{resource-name}/{version}/{resource}.yaml
+
+# The README of the versioned resource (usage, …)
+./{resource-type}/{resource-name}/{version}/README.md
+
+# Optional: Test working samples.
+# Those samples would be used to run automated tests
+./{resource-type}/{resource-name}/{version}/tests/…
+# Optional: Addition samples.
+# Those samples would be used to run automated tests
+./{resource-type}/{resource-name}/{version}/samples/… |
+```
+
+
+For example (with Task and Pipelines):
+
+```
+./task/
+  /argocd
+    /0.1
+      /README.md
+      /argocd.yaml
+      /samples/deploy-to-k8s.yaml
+    /0.2/...
+    /OWNERS
+    /README.md
+  /golang-build
+    /0.1
+      /README.md
+      /golang-build.yaml
+      /samples/golang-build.yaml
+./pipelines/
+  /go-release
+    /0.1
+      /README.md
+      /go-release.yaml
+      /samples/dummy-go-release.yaml |
+```
+
+An example of a *freezed* catalog migrated to this layout is available [here](https://github.com/Pipelines-Marketplace/catalog).
+
+#### Open questions
+
+1. **Flat hierarchy or categories ?**
+What about grouping resources, an example would be some Tasks, related to github. We may have a bunch of github related tasks (github-create-pullrequest, github-create-comment, github-update-check, …). Should those be grouped or "flat" ?
+
+### Requirements & Guidelines
+
+Guidelines aim to help users author their tekton resource. Requirements are *applying* those requirements to the upstream catalog. The requirements are not part of the Catalog contract but they should be applied as much as possible to other catalogs.
+
+Automation might be run against all verified and official resources to ensure (see [Tekton Catalog Test Infrastructure](https://docs.google.com/document/d/1-czjvjfpuIqYKsfkvZ5RxIbtoFNLTEtOxaZB71Aehdg/edit#)):
+
+1. Resources have all description fields filled in
+
+2. Resources must have app.kubernetes.io /version to indicate the version of the resource
+
+3. Resources can include annotation.tags which is a comma separated tags associated with it e.g. kaniko task can have a tag: container-image, build-tool
```

Comment:
can we make these `tekton.dev/tags`? 

https://github.com/tektoncd/catalog/pull/389 mentions `displayName` as well, same suggestion there

+ @PuneetPunamiya

Reply:
These are 2 minor changes that we can incorporate, before #tekoncd/catalog reorg merges - https://github.com/tektoncd/catalog/pull/389 . Hub preview will need some work but I think we should use tekton specific annotations.

Reply:
Updated